toolkit/tools/scheduler/scheduler.go

```diff
@@ -457,6 +476,53 @@ func updateGraphWithImplicitProvides(res *schedulerutils.BuildResult, pkgGraph *
 	return
 }
 
+// setAssociatedDeltaPaths sets the DeltaPath for all of the request's ancillary nodes (both build and run) to the actual
+// RPM paths. A delta node will normally point at the cached RPM path, but we want to point it at the actual RPM if we built it.
+func setAssociatedDeltaPaths(res *schedulerutils.BuildResult, builtFiles []string, pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex) (err error) {
+	graphMutex.Lock()
+	defer graphMutex.Unlock()
+
+	if !res.WasDelta {
```

Comment:
>if !res.WasDelta { [](http://example.com/codeflow?start=1&length=18)

Since this is an internal function (so we have full control over the input) and we're already making sure we only pass results that were created for delta nodes, there's no need to re-do this check here. I'd either remove it from here, or move the whole `res.Node.Type == pkggraph.TypeBuild && res.WasDelta` check into this function, leave early if that statement is fasle, and potentially re-name the function itself. I think the former is a bit cleaner.

Reply:
Also, I guess passing a result for a non-build node would also be an error in the context of this function.

Reply:
Gone now.